### PR TITLE
[tune] document scalability best practices (k8s, scalability thresholds)

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/api_docs/scalability.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
+.. _tune-scalability:
+
+Scalability and overhead benchmarks
+===================================
+
+We conducted a series of microbenchmarks where we tested the scalability of Ray Tune and tried to reason about
+performance overhead we observed. The results from these benchmarks are reflected in the documentation,
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:7]:
```suggestion
We conducted a series of microbenchmarks where we evaluated the scalability of Ray Tune and analyzed the
performance overhead we observed. The results from these benchmarks are reflected in the documentation,
```

File: doc/source/tune/api_docs/scalability.rst [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+e.g. when we make suggestions on :ref:`how to remove performance bottlenecks <tune-bottlenecks>`.
+
+This page gives an overview over the experiments we did. For each of these experiments, the goal was to
+examine the total runtime of the experiment and address issues when the observed overhead compared to the
+minimal theoretical time was too high (e.g. more than 20% overhead).
+
+
+.. list-table:: TrialScheduler Feature Compatibility Matrix
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:15]: change table name?

File: doc/source/tune/api_docs/scalability.rst [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+===================================
+
+We conducted a series of microbenchmarks where we tested the scalability of Ray Tune and tried to reason about
+performance overhead we observed. The results from these benchmarks are reflected in the documentation,
+e.g. when we make suggestions on :ref:`how to remove performance bottlenecks <tune-bottlenecks>`.
+
+This page gives an overview over the experiments we did. For each of these experiments, the goal was to
+examine the total runtime of the experiment and address issues when the observed overhead compared to the
+minimal theoretical time was too high (e.g. more than 20% overhead).
+
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:13]: Maybe you want to note here that you disabled syncing/extra loggers for these experiments?

File: doc/source/tune/api_docs/scalability.rst [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+     - 300
+     - 392.42
+
+
+Below we discuss some insights on results where we observed much overhead.
+
+
+Result throughput
+-----------------
+Result throughput is limited by the time it takes to process results. When a trial reports results, it only
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:84]: Maybe start with `What is result throughput`?

File: doc/source/tune/api_docs/scalability.rst [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+results.
+
+To speed the process up, Ray Tune adaptively buffers results, so that trial training is continued earlier if
+many trials are running in parallel and report many results at the same time. Still, processing hundreds of
+results per trial for dozens or hundreds of trials can become a bottleneck.
+
+**Main insight**: Ray Tune will throw a warning when trial processing becomes a bottleneck. If you notice
+that this becomes a problem, please follow our guidelines outlined :ref:`in the FAQ <tune-bottlenecks>`.
+Generally, it is advised to not report too many results at the same time. Consider increasing the report
+intervals.
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:96]: Consider increasing the report intervals by ...

File: doc/source/tune/api_docs/scalability.rst [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
++-------------+--------------------------+---------+---------------+------------------+---------+
+
+
+Many results on a single node
+"""""""""""""""""""""""""""""
+In this setup, loggers (CSV, JSON, and TensorboardX) are disabled, except when
+explicitly noted.
+
+On a single node, throughput seems to be a bit higher. With logging, handling 1000 results per second
+seems acceptable in terms of overhead, though you should probably still target for a lower number.
```
Comment (richardliaw) [inline doc/source/tune/api_docs/scalability.rst:139]: Can you describe the experiment you ran here? Something like:

```
In this experiment, we measured the scalability of the system under different trial reporting frequencies (changing the number of results per second per trial).

**Review by richardliaw — COMMENTED** [review]
Left some changes!

File: doc/source/tune/api_docs/scalability.rst [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+that this becomes a problem, please follow our guidelines outlined :ref:`in the FAQ <tune-bottlenecks>`.
+Generally, it is advised to not report too many results at the same time. Consider increasing the report
+intervals.
+
+Below we present more detailed results on the result throughput performance.
+
+Many concurrent trials
+""""""""""""""""""""""
+In this setup, loggers (CSV, JSON, and TensorboardX) and trial synchronization are disabled, except when
+explicitly noted.
```
[the remark this answers is not in the document]
Comment (krfricke) [reply doc/source/tune/api_docs/scalability.rst:103]: I'll keep the notes here but added a note at the table as well

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Lint failing?

**Pull request merged**